apps/api/src/form/form.controller.ts

```diff
@@ -43,4 +50,39 @@ export class FormController {
   async getFormById(@Param('id') id: string): Promise<FormEntity> {
     return await this.formService.getFormById(id);
   }
+
+  @ApiOperation({
+    summary: 'Export Data',
+  })
+  @UseInterceptors(ClassSerializerInterceptor)
+  @ApiResponse({ status: HttpStatus.CREATED, type: EmptyResponse })
+  @HttpCode(HttpStatus.CREATED)
+  @Get('/export/:passCode')
+  async exportAll(@Param('passCode') passCode: string, @Res() res: Response) {
+    // TODO: Add logging
+
+    // TODO: Based on auth roles in the future.
+    if (passCode === process.env.EXPORT_SECRET) {
+      res.set({
+        'Content-Type': 'text/csv',
+        'Content-Disposition': 'attachment;filename=ehrp.csv',
+      });
+
+      // NOTE: Could be paginated
+      // Current Metric ~ 150K records ~ 3 sec ~ 38 MB
```

Comment:
Napkin math means we can support 23k records before hitting the 6mb lambda payload limit? Should be fine for now

Reply:
That's the payload limit - NOT the download to client limit. We have no payload in this case.

Reply:
Interesting, I thought we'd faced issues with that in the past, I could be wrong

Reply:
![image](https://user-images.githubusercontent.com/67800715/145471677-be7be3ba-2b47-4b36-a7a3-8fd37fa23195.png)

Reply:
Maybe I'm confused, why isn't it a payload?

Reply:
AH! that's bad. Okay, I think still it's fine, until we convert to storing this into S3. 

Napkin math works for now !

Reply:
I meant request has no payload.

Reply:
Yep sounds good, should be a while before we hit 6mb. I'll add a ticket to the backlog